### ISSUE TITLE
Support Restify

### ISSUE THREAD

**Issue by Gby56** [noir-cr/noir#272]
Hi ! I was checking the list of supported frameworks, seems like Restify is not there yet, I might make a PR later :) 
http://restify.com/

**Comment by Gby56** [comment]
@hahwul I'm not familiar at all with Crystal but I'm getting an error when building, even from scratch
```
shards build --error-trace

In src/deliver/send_proxy.cr:22:26

 22 | Crest::Request.execute(
                     ^------
Error: instantiating 'Crest::Request.execute(method: Symbol, url: String, p_addr: (String | Nil), p_port: (Int32 | Nil), tls: OpenSSL::SSL::Context::Client, user_agent: String, params: Hash(String, String), headers: Hash(String, String), form: Hash(String, String), json: Bool)'
.....

In lib/http_proxy/src/ext/http/client.cr:11:28

 11 | @io = proxy_client.open(
                         ^---
Error: no overload matches 'HTTP::Proxy::Client#open', host: String, port: Int32, tls: (OpenSSL::SSL::Context::Client | Nil), dns_timeout: (Time::Span | Nil), connect_timeout: (Time::Span | Nil), read_timeout: (Time::Span | Nil), write_timeout: (Time::Span | Nil)

Overloads are:
 - HTTP::Proxy::Client#open(host, port, tls = nil, *, dns_timeout : ::Float64 | ::Nil, connect_timeout : ::Float64 | ::Nil, read_timeout : ::Float64 | ::Nil, write_timeout : ::Float64 | ::Nil)
Couldn't find overloads for these types:

```

**Comment by hahwul** [comment]
Hiya @Gby56

Thank you so much for your wonderful ideas and contributions!

I apologize for the late response as I am currently on a restful vacation 😭

I've taken a quick look at the PR you sent. I'll be on vacation until next week, but once I'm back, I'll roll up my sleeves and do my best to ensure your suggestions can be integrated into the main code!

Thanks again! Have a great day.

**Comment by hahwul** [comment]
Released [v0.15.0](https://github.com/noir-cr/noir/releases/tag/v0.15.0). Closing the issue.
